packages/astro/src/core/ssr/resolve-config-paths.ts

```diff
@@ -0,0 +1,94 @@
+import { tsconfigResolverSync, TsConfigResultSuccess } from 'tsconfig-resolver';
+import fs from 'fs'
+import { posix as ps } from 'path'
+import type * as vite from 'vite'
+
+/** Return the nearest tsconfig.json or jsconfig.json configuration result. */
+const getConfigResult = (): TsConfigResultSuccess | null => {
+	/** Closest tsconfig.json */
+	const tsConfig = tsconfigResolverSync({ searchName: 'tsconfig.json' })
+
+	// return tsconfig.json, if it exists
+	if (tsConfig.exists) return tsConfig
+
+	/** Closest jsconfig.json */
+	const jsConfig = tsconfigResolverSync({ searchName: 'jsconfig.json' })
+
+	// return jsconfig.json, if it exists
+	if (jsConfig.exists) return jsConfig
+
+	// return null if no config exists
+	return null
+}
+
+/** Return the resolvers from the tsconfig.json or jsconfig.json configuration. */
+const getConfigResolver = (): { baseUrl: string, paths: { [alias: string]: string[] } } | null => {
+	/** Configuration result */
+	const configResult = getConfigResult()
+
+	if (configResult && configResult.config.compilerOptions?.baseUrl) {
+		let { baseUrl, paths } = configResult.config.compilerOptions as {
+			baseUrl: string
+			paths: {
+				[key: string]: string[];
+			}
+		}
+
+		// resolve the base url from the configuration file directory
+		baseUrl = ps.resolve(ps.dirname(configResult.path), baseUrl)
+
+		// guarantee paths is an object
+		paths = Object(paths)
+
+		// sanitize and resolve all aliases from the configuration file directory
+		for (let [ alias, values ] of Object.entries(paths)) {
+			delete paths[alias]
+
+			paths[alias.replace(/\*$/, '')] = values = [].concat(values as any).map(
+				(value: string) => ps.resolve(baseUrl, value).replace(/\*$/, '')
+			)
+		}
+
+		return { baseUrl, paths }
+	}
+
+	return null
+}
+
+export default function resolveConfigPaths(): vite.PluginOption {
+	/** Resolver from the tsconfig.json or jsconfig.json configuration. */
+	const resolver = getConfigResolver()
+
+	if (!resolver) return {} as vite.PluginOption
+
+	return {
+		name: 'resolve-config-paths',
+		enforce: 'pre',
+		resolveId(source: string): string | null {
```

Comment:
`source` here is not the source code, but the `id` or `specifier`, right? Might rename this for clarity

Reply:
What types of files is this resolving? Note that whatever `resolveID()` claims will get excluded from other plugins.

Reply:
I named this `source` to match the [documentation for `resolveId`](https://rollupjs.org/guide/en/#resolveid), but I’m fine with renaming this. Would you give this a thumbs up if you want it renamed, and if so then I’ll do that?

Reply:
> What types of files is this resolving?

Any _importee_ run thru [rollup](https://rollupjs.org/guide/en/#resolveid).

Reply:
In that case, using `fs` checks might cause a problem! Can we use `this.resolve` like the example does?

Reply:
Done in configAliasVitePlugin

Reply:
Done in dd26e2e34b703849baeaf3d5c60e6d3da175d54b